src/isar/apis/schedule/start_mission.py

```diff
@@ -41,14 +44,23 @@ def post(
         except MissionPlannerError as e:
             message = StartMissionMessages.mission_not_found()
             self.logger.error(e)
-            return JSONResponse(
-                content=asdict(message), status_code=HTTPStatus.NOT_FOUND
-            )
+            response.status_code = HTTPStatus.NOT_FOUND.value
+            return StartResponse(message=message.message, started=message.started)
 
-        ready, response = self.scheduling_utilities.ready_to_start_mission()
+        ready, response_scheduler = self.scheduling_utilities.ready_to_start_mission()
         if ready:
-            response = self.scheduling_utilities.start_mission(mission=mission)
-            self.logger.info(response)
+            response_scheduler = self.scheduling_utilities.start_mission(
+                mission=mission
+            )
+            self.logger.info(response_scheduler)
 
-        message, status_code = response
-        return JSONResponse(content=asdict(message), status_code=status_code)
+        message, status_code = response_scheduler
+        response.status_code = status_code.value
```

Comment:
For example this line; I don't see where response is used after this, so why is the status_code attribute set here?

Reply:
This is the way to set the status code in the fastapi. If this is not set we will return 200 in all cases. This is due to using pydantic models, which gives better documentation in the swagger docs, as return type opposed to using a JSONResponse as previous. See [fastapi](https://fastapi.tiangolo.com/advanced/response-change-status-code/)